src/Parser/EnhancementShaman/CombatLogParser.js

```diff
@@ -0,0 +1,336 @@
+
+    const abilityTracker = this.modules.abilityTracker;
+    const getAbility = spellId => abilityTracker.getAbility(spellId);
+
+    const flametongue = getAbility(SPELLS.FLAMETONGUE.id);
+    const frostbrand = getAbility(SPELLS.FROSTBRAND.id);
+    const stormBringer = getAbility(SPELLS.STORMBRINGER.id);
+
+    const fightDuration = this.fightDuration;
+
+	  //uptimes
+	  const flametongueUptime = this.selectedCombatant.getBuffUptime(SPELLS.FLAMETONGUE_BUFF.id) / this.fightDuration;
+	  const frostbrandUptime = this.selectedCombatant.getBuffUptime(SPELLS.FROSTBRAND.id) / this.fightDuration;
+    const landslideUptime = this.selectedCombatant.getBuffUptime(SPELLS.LANDSLIDE_BUFF.id) / this.fightDuration;
+    const furyofairUptime = this.selectedCombatant.getBuffUptime(SPELLS.FURY_OF_AIR.id) / this.fightDuration;
+
+    const nonDpsTimePercentage = this.modules.alwaysBeCasting.totalDamagingTimeWasted / fightDuration;
+    const deadTimePercentage = this.modules.alwaysBeCasting.totalTimeWasted / fightDuration;
+
+    const masteryPercent = this.selectedCombatant.masteryPercentage;
+    const hastePercent = this.selectedCombatant.hastePercentage;
+
+    //Legendaries
+    // const hasUncertainReminder = this.selectedCombatant.hasHead(ITEMS.UNCERTAIN_REMINDER.id);
+    // const hasEmalons = this.selectedCombatant.hasChest(ITEMS.EMALONS_CHARGED_CORE.id);
+    const hasAkainus = this.selectedCombatant.hasWrists(ITEMS.AKAINUS_ABSOLUTE_JUSTICE.id);
+    // const hasSmolderingHeart = this.selectedCombatant.hasHands(ITEMS.SMOLDERING_HEART.id);
+    // const hasStorm = this.selectedCombatant.hasWaist(ITEMS.STORM_TEMPESTS.id);
+    // const hasRoots = this.selectedCombatant.hasLegs(ITEMS.ROOTS_OF_SHALADRASSIL.id);
+    // const hasSpiritual = this.selectedCombatant.hasFeet(ITEMS.SPIRITUAL_JOURNEY.id);
+    // const hasSoul = this.selectedCombatant.hasFinger(ITEMS.SOUL_OF_THE_FARSEER.id);
+    const hasEOTN = this.selectedCombatant.hasFinger(ITEMS.EYE_OF_THE_TWISTING_NETHER.id);
+    // const hasSephuz = this.selectedCombatant.hasFinger(ITEMS.SEPHUZS_SECRET.id);
+
+  this.selectedCombatant._combatantInfo.gear.forEach(function(value) {
+    const equippedItem = getItemInfo(value.id);
+
+    if(equippedItem !== undefined && equippedItem.quality === 5) {
+      results.items.push({
+        item: equippedItem,
+        result: (
+          <dfn data-tip="">
+            Equipped Legendary
+          </dfn>
+        ),
+      });
+    }
+  });
+
+  //add trinkets to item list
+  function getItemInfo(id) {
+    let value;
+    Object.keys(ITEMS).some(function(k) {
+      if(ITEMS[k].id === id) {
+        value = ITEMS[k];
+      }
+    });
+
+    return value;
+  }
+
+    if (nonDpsTimePercentage > 0.3) {
+      results.addIssue({
+        issue: `[NYI] Your non DPS time can be improved. Try to cast damaging spells more regularly (${Math.round(nonDpsTimePercentage * 100)}% non DPS time).`,
```

Comment:
I recently changed the way suggestions should be provided;
The part of the suggestion that shows the actual performance (usually at the end of the sentence in parenthesis) should now go in the `stat` property of an issue, and it should now include a recommended target that is based on the breakpoint that shows the suggestion (i.e. the minor importance level).
This diff shows the changes that were needed for Holy Paladins:
https://github.com/MartijnHols/WoWAnalyzer/commit/035d4cae5d94e52cad1b624b2649bd921d09cdbd#diff-16d75c2e452c4b58bec847e9818c57f0L233

While existing specs are not required to migrate right away, new suggestions should do this unless there's a good reason not to. Sorry you got caught up between this change.

This goes for all further suggestions as well.